### ISSUE TITLE
Selections Cursortip error

### ISSUE THREAD

**Issue by sprunk** [ZeroK-RTS/Zero-K#2570]
```
[f=0006294] [cawidgets.lua] Error: Error in Update(): [string "LuaUI/Widgets/gui_chili_selections_and_curs..."]:1831: attempt to index local 'ud' (a nil value)
[f=0006294] [cawidgets.lua] Error: Removed widget: Chili Selections & CursorTip v2
```
https://github.com/ZeroK-RTS/CrashReports/issues/1977

**Comment by GoogleFrog** [comment]
For these bugs fix and make a stable or revert the engine if appropriate.

**Comment by sprunk** [comment]
Bad timing. This spam of "X error" is unrelated to the new engine (it coincided with me digging through crash reports).

**Comment by Sgathec** [comment]
I would like to fix this bug. I need to fix one due to an assignment.

**Comment by sprunk** [comment]
I've tried to see where the invalid unitDefID actually comes from but there is nothing obvious in the code.

I assumed it comes from selecting enemy units in the fog with `/specfullview 2` (can select but can't see) because there's a different part of the code which assigns such unknown units the invalid unitDefID 0 (as opposed to nil) but I couldn't reproduce or see how it would propagate.

For now I also did a naive check.
